langtest/datahandler/datasource.py

```
@@ -94,14 +94,14 @@ class DataFactory:
     correct Dataset type based on the file extension.
     """
 
-    def __init__(self, file_path: str, task: str, **kwargs) -> None:
+    def __init__(self, file_path: dict, task: str, **kwargs) -> None:
         """Initializes DataFactory object.
 
         Args:
-            file_path (str): Path to the dataset.
+            file_path (dict): Dictionary containing 'data_source' key with the path to the dataset.
             task (str): Task to be evaluated.
         """
-        self._file_path = file_path
+        self._file_path = file_path.get("data_source")
```

if **data_source** is not provided, then it should raise an error.

---

please make it changes, @Prikshit7766 @RakshitKhajuria